shim/src/main.rs

```diff
@@ -33,8 +55,58 @@ fn run_current_turbo(args: Vec<String>) -> Result<i32> {
     Ok(exit_code)
 }
 
+/// Checks if we are in single package mode by first seeing if there is a turbo.json
+/// in the ancestor path, and then checking for workspaces.
+///
+/// # Arguments
+///
+/// * `current_dir`: Current working directory
+///
+/// returns: Result<bool, Error>
+///
+fn is_single_package_mode(current_dir: &Path) -> Result<bool> {
+    let has_turbo_json = AncestorSearch::new(current_dir.to_path_buf(), "turbo.json")?
+        .next()
+        .is_some();
+
+    if has_turbo_json {
+        return Ok(false);
+    }
+
+    // We should detect which package manager and then determine workspaces from there,
+    // but detection is not implemented yet and really we're either checking the `package.json`
+    // or the `pnpm-workspace.yaml` file so we can do both.
+    let npm = PackageManager::Npm;
+    if npm.get_workspace_globs(current_dir).is_ok() {
+        return Ok(false);
+    };
+
+    let pnpm = PackageManager::Pnpm;
+    if pnpm.get_workspace_globs(current_dir).is_ok() {
+        return Ok(false);
+    };
+
+    Ok(true)
+}
+
+fn is_run_command(clap_args: &Args) -> bool {
```

Comment:
This is unfortunately more complicated. Anything that is not otherwise defined is also an implicit `run`. So `turbo build` is actually treated as `turbo run build`

Reply:
Ahh gotcha. Well I can parse out the rest of the subcommands